src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -2645,12 +2936,8 @@ void processContextElement
   //
   // FIXME P6: Once we allow for ServicePath to be modified, this loop must be looked at.
   //
-  int docs = 0;
-
-  // Used to accumulate error response information, checked at the end
-  bool         attributeAlreadyExistsError = false;
-  std::string  attributeAlreadyExistsList  = "[ ";
 
+  std::vector<BSONObj> results;
   while (moreSafe(cursor))
```

Comment:
So, you invented this 'two phased op' to not holding the connection for too long, right?
To me it seems like a good idea; connections will be held longer now, that we know about the problem with cursors.

About using this approach in all cases ... Yes, why not? But, as a separate issue about throughput. It's all about minimizing the time to wait for a mongo connection, right?